web/WEB-INF/web.xml

```diff
@@ -14,9 +14,41 @@
     <description>An address where OpenGrok can receive new configuration</description>
   </context-param>
 
+  <!--
+  <context-param>
+      <param-name>authorizationPluginDirectory</param-name>
+      <param-value>/var/opengrok/plugins</param-value>
+      <description>
+          Directory with authorization plugins.
+          Default value is dataRoot/../plugins 
+            (can be /var/opengrok/plugins if
+             dataRoot is /var/opengrok/data)
+      </description>
+  </context-param>
+  
+  <context-param>
+      <param-name>enableAuthorizationWatchDog</param-name>
+      <param-value>true</param-value>
+      <description>
+          Enable watching the plugin directory for changes in real time.
+          Suitable for development.
+      </description>
+  </context-param>
+  -->
+  
   <listener>
    <listener-class>org.opensolaris.opengrok.web.WebappListener</listener-class>
   </listener>
+
```

Comment:
The filter itself has almost no impact.

The implementation has:
 - checking for project is linear in size of projects
 - if there is such a project then it goes (number of plugins * number of projects)

Further impact depends on the plugin implementations. 
There can be:
 - another linear time for checking allowed projects
 - extra time for creating session if needed
 - another extra time for special calls of the plugin

The framework ensures that for a single request there will be no more than one call to a plugin isAllowed per single project/group (with attribute cache).